### DISCUSSION TITLE
Cannot duplicate flows or copy blocks between.  Bonus - make Flows modules.

### DISCUSSION THREAD

**Discussion by oehm-smith** [postmanlabs/postman-flows#25] [Ideas]
I'm writing tests using Flows that have parts that are the same and I'd like to just be able to duplicate a Flow or at worst copy blocks between.  This functionality is not currently supported but would be nice to see it.

It would also nice to be able to create modules so the parts that are common could be made in to modules and dropped in as a Block.

Overall, well done on this feature - it works quite well as it currently is (13.10.21) and is not just for NoCoders.

**Comment by samay-v** [comment]
That is some great feedback @oehm-smith!
We're currently exploring the idea of what you called "modules" as a way to encapsulate and reuse logic built with flow, in flow.
I'll create an enhancement issue to track the Copy between flows feature you mentioned. 

https://github.com/postmanlabs/postman-flows/issues/28#issue-1024765601

**Reply by oehm-smith** [reply]
Sweet nice to hear and thanks

**Comment by wmaynard** [comment]
Bumping this because I would really like to be able to duplicate Flows.

**Comment by tewley** [comment]
I was surprised to learn flows can't be duplicated. This would be a very useful feature.

**Reply by saswatds** [reply]
Hey @tewley , we will add this feature soon.

**Comment by saswatds** [comment]
Hey folks, I have a question. If we enable using Flows as modules, would you still want to duplicate flows?

**Comment by wmaynard** [comment]
While I would have less need with the ability to use flows as modules, I think it still makes sense to duplicate flows.  I can duplicate collections and environments, so I would expect to be able to duplicate flows as well.

**Reply by hishamMounir** [reply]
May be we can duplicate the local files that belongs to flows stored on the local machine but with different names

**Comment by melbatoastie** [comment]
I love the look and feel of flows though I did struggle to get my first basic one working. However, without the ability to copy flows I have parked my usage of flows for the following reasons:
1. I don't want to break the Flow I have developed while trying to extend it.
2. I want to re-use much of the logic from my existing flow
3. Without such a basic feature, it feels as though Postman Flows may not become a finished product so it would be risky to put more effort into it.
Is there any ETA for when this feature OR a big release is due for Flows?

**Reply by saswatds** [reply]
Hey @melbatoastie, This quarter is all about improving what you have pointed out.
1. We have already merged the ability to duplicate a Flow, it should be out in the next release
2. We have been working on a spec to enable re-use of flows inside other flows. We are internally testing it and ship soon gradually.

Expect that you would see a lot of UX improvement and last-mile finishing of the product in this quarter.

**Reply by melbatoastie** [reply]
Thanks. That sounds ideal.
